EMS/core-bundle/src/Repository/EnvironmentRepository.php

```diff
@@ -97,6 +97,67 @@ public function getDeletedRevisionsPerEnvironment(Environment $environment): int
         }
     }
 
+    public function orderKeyExists(int $orderKey): bool
+    {
+        $environmentsStats = $this->getEnvironmentsStats();
+        foreach ($environmentsStats as $environmentStats) {
+            $environmentOrderKey = $environmentStats['environment']->getOrderKey();
+            if ($environmentOrderKey === $orderKey) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public function getMaxOrderKey(): int
+    {
+        $qb = $this->createQueryBuilder('environment');
+
+        $qb->select('MAX(environment.orderKey) as max_order_key');
+
+        try {
+            $result = $qb->getQuery()->getSingleScalarResult();
+
+            return $result ? (int) $result : 0;
+        } catch (NonUniqueResultException $e) {
+            // Handle exception if needed
+            return 0;
+        }
+    }
+
+    public function updateOrderKeysAfterDelete(int $deletedOrderKey): void
+    {
+        $qb = $this->createQueryBuilder('e');
+        $qb->update()
+            ->set('e.orderKey', 'e.orderKey - 1')
+            ->where($qb->expr()->gt('e.orderKey', $deletedOrderKey))
+            ->getQuery()
+            ->execute();
+    }
+
```

Comment:
I don't get it

Reply:
In the previous version, the order number is a list displayed in the view and the orderKeys are not updated, here orderKeys are displayed

Reply:
not needed, less code is better